src/components/shared/StepWidget.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import { TiTick } from 'react-icons/ti';
+import '../../assets/styles/components/stepper.scss';
+
+type stepperProps = { steps: string[] };
+
+function Stepper({ steps }: stepperProps) {
+  const [currentStep, setCurrentStep] = useState(1);
+  const [complete, setComplet] = useState(false);
+
+  function next() {
+    if (currentStep === steps.length) setComplet(true);
+    else {
+      setCurrentStep((prev: number) => prev + 1);
+    }
+  }
+
+  function previous() {
+    if (currentStep > 1) {
+      setComplet(false);
+      setCurrentStep((prev: number) => prev - 1);
+    }
+  }
+
+  return (
+    <div className="absolute top-[10vh] left-[0vw] flex sm:flex-row flex-col  justify-between">
+      {steps?.map((step, i) => (
+        <div
+          key={i}
```

Comment:
This is where you got the error. You were trying to use an index as the key, which is not very wise, because there might be instances (though rare) where the keys might match for some given context.

I would suggest you come up with some creative ways to generate some random ID as it as the key.